pathd/path_pcep_cli.c

```diff
@@ -1356,34 +1541,61 @@ static void print_pcep_session(struct vty *vty, struct pce_opts *pce_opts,
 }
 
 static int path_pcep_cli_show_srte_pcep_session(struct vty *vty,
-						const char *pcc_peer)
+						const char *pcc_peer, bool uj)
 {
 	struct pce_opts_cli *pce_opts_cli;
 	struct pcep_pcc_info *pcc_info;
+	json_object *json = NULL;
+
+	if (uj)
+		json = json_object_new_object();
 
 	/* Only show 1 PCEP session */
 	if (pcc_peer != NULL) {
+		if (json)
+			json_object_string_add(json, "pceName", pcc_peer);
 		pce_opts_cli = pcep_cli_find_pce(pcc_peer);
 		if (pce_opts_cli == NULL) {
-			vty_out(vty, "%% PCE [%s] does not exist.\n", pcc_peer);
+			if (json)
+				json_object_string_add(json, "warning",
+						       "PCE does not exist.");
+			else
+				vty_out(vty, "%% PCE [%s] does not exist.\n",
+					pcc_peer);
 			return CMD_WARNING;
```

Comment:
We should call `vty_json()`, otherwise memory leak for `json`, right?

Reply:
right. updated.